### ISSUE TITLE
Yara 3.9.0 hex pattern issue

### ISSUE THREAD

**Issue by Tim-Sobolev** [VirusTotal/yara#1025]
Hex pattern is OK with Yara 3.8 and previous versions.
It is not working on Yara 3.9 (Python module and provided command-line version).

[test_yara_390.zip](https://github.com/VirusTotal/yara/files/2904875/test_yara_390.zip)

**Comment by chaplja** [comment]
Looks like trailing ?? is not handled well. Ruleset is properly matched if you remove the ?? on the end.

**Comment by Tim-Sobolev** [comment]
After adding after trailing ?? one more actual byte (E8) Yara acts in the same manner.

**Comment by aprokofiev** [comment]
Trailing masks do not match alright but at least it could be considered logical, since a trailing `??` does not add much precision anyway...
```yara
rule trailing_1 {strings: $a = { 0f 95 ?? a2 fd e3 ?? } condition: $a}
rule trailing_2 {strings: $a = { 0f 95 ?? ?2 fd e3 ?? } condition: $a}
```
...notice however that these two will work nevertheless:
```yara
rule valid_example_1 {strings: $a = { 0f 95 c0 a2 fd e3 ?? } condition: $a}
rule valid_example_2 {strings: $a = { 0f 9? c? a? f? ?? ?? } condition: $a}
```
Sadly enough partial masks are broken too:
```yara
rule partial_1 {strings: $a = { 0f ?5 ?? a2 fd ?? ?1 } condition: $a}
rule partial_2 {strings: $a = { 0f ?5 ?? a2 fd ?? 4? } condition: $a}
```
But the most important thing is - these rules are also do not match:
```yara
rule character_1 {strings: $a = { 0f ?? c0 a2 fd ?? 41 } condition: $a}
rule character_2 {strings: $a = { 0f ?? c0 ?2 fd ?? 41 } condition: $a}
rule character_3 {strings: $a = { 0f ?? c0 a? fd ?? 41 } condition: $a}
```
Notice that there are no masks at the end whatsoever.

Looks like the whole deal was broken with this 5f1912fb2b6c0c0c4246d30d216a7b49bbf1f2d3 commit by @plusvic (I built it on Linux using `./configure --disable-shared --enable-static`) - the previous one (cf1d767184f7f6ce8a00fbfdde5866418f3a4628) is working just fine.

P.S. I used the very same `test.bin` file @Tim-Sobolev attached to his initial message, but you could easily truncate it to first 7 bytes and the whole thing will still remain valid: `0f 95 c0 a2 fd e3 41`.

**Comment by plusvic** [comment]
Fixed in 46a54d2e404622be79db922f8e80176e356ab13d.
